jsapp/js/account/stripe.api.ts

```diff
@@ -120,17 +119,52 @@ export async function getSubscriptionInterval() {
   return 'month';
 }
 
+const defaultLimits: AccountLimit = Object.freeze({
+  submission_limit: 'unlimited',
+  nlp_seconds_limit: 'unlimited',
+  nlp_character_limit: 'unlimited',
+  storage_bytes_limit: 'unlimited',
+});
+
+function getLimitsForMetadata(
+  metadata: {[key: string]: string},
+  limitsToCompare: false | AccountLimit = false
+) {
+  const limits: Partial<AccountLimit> = {};
+  for (const [key, value] of Object.entries(metadata)) {
+    // if we need to compare limits, make sure we're not overwriting a higher limit from somewhere else
+    if (limitsToCompare) {
+      if (
+        !limitsToCompare?.[key as keyof AccountLimit] ||
+        (value !== 'unlimited' &&
```

Comment:
I don't like that I can add a typo here and the code compiles. value ends up as a string and not a number | unlimited. I'm tempted to say that unlimited should be represented as a null, enum, variable, or improve the types somehow. I'll leave this to you to act upon or not. I think I prefer null because "Your limit is null" seems like a correct thing to say, from a weird programmer context at least. But changing to null may be a too large of a change to consider.

If you feel that value could in reality contain weird stuff like "unlimitted" then a simple solve would just be to make something like `const UNLIMITED_TOKEN = "unlimited";` to prevent the programmer from making a mistake.